### ISSUE TITLE
odo delete doesn’t take in account of context when deleting devfile.yaml

### ISSUE THREAD

**Issue by girishramnani** [openshift/odo#4527]
/kind bug

## What versions of software are you using?

**Operating System:**

**Output of `odo version`: odo master

## Actual behavior
```
(base) girish@MacBook-Pro ~ % odo delete --context /var/folders/b8/r_v6zg3x2x9c7mpf6csnxs4h0000gn/T/375762719 -f --all
Gathering information for component nodejs-x375762719-dfyl
 ✗  Checking status for component [301ms]
 ⚠  pod not found for the selector: component=nodejs-x375762719-dfyl
Deleting local config
 ✓  Successfully deleted env file
 ✗  devfile.yaml does not exist in the current directory
```
folder content
```
ls /var/folders/b8/r_v6zg3x2x9c7mpf6csnxs4h0000gn/T/375762719
config          devfile.yaml    package.json    server.js
```

## Expected behavior
devfile should be deleted

**Comment by girishramnani** [comment]
working on this

**Comment by dharmit** [comment]
Moving to Sprint 199 since this won't get closed till the PR #4518 (which also covers other things, IIUC) gets merged.
